tests/auto_run_tests.pl

```diff
@@ -2,60 +2,228 @@
     & eval 'exec perl -S $0 $argv:q'
     if 0;
 
-# -*- perl -*-
-# This file is for running the run_test.pl scripts listed in
-# auto_run_tests.lst.
+# Executes test list files (*.lst), which contain commands with conditions
+# called configurations under which the commands are run.
 
-use lib "$ENV{ACE_ROOT}/bin";
-use lib "$ENV{DDS_ROOT}/bin";
+use strict;
+use warnings;
+
+use Env qw(ACE_ROOT DDS_ROOT PATH);
+
+use lib "$ACE_ROOT/bin";
+use lib "$DDS_ROOT/bin";
 use PerlDDS::Run_Test;
 
 use Getopt::Long;
 use Cwd;
-use Env qw(DDS_ROOT PATH);
-use strict;
+use POSIX qw(SIGINT);
+
+my $dry_run = 0;
+
+sub cd {
+    my $dir = shift;
+    chdir($dir) || die "auto_run_tests: Error: Cannot chdir to $dir";
+}
+
+sub run_command {
+    my $what = shift;
+    my $command = shift;
+    my $print_error = shift // 1;
+
+    if ($dry_run) {
+        my $cwd = getcwd();
+        print "In \"$cwd\" would run:\n    $command\n";
+        return 0;
+    }
+
+    my $result = 0;
+    if (system($command)) {
+        $result = $? >> 8;
+        my $signal = $? & 127;
+        die("auto_run_tests: \"$what\" was interrupted") if ($signal == SIGINT);
+        if ($print_error) {
+            my $coredump = $? & 128;
+            my $error_message;
+            if ($? == -1) {
+                $error_message = "failed to run: $!";
+            }
+            elsif ($signal) {
+                $error_message = sprintf("exited on signal %d", ($signal));
+                $error_message .= " and created coredump" if ($coredump);
+            }
+            else {
+                $error_message = sprintf("returned with status %d", $result);
+            }
+            print "auto_run_tests: Error: $what $error_message\n";
+        }
+    }
+    return $result;
+}
+
+my @builtin_test_lists = (
+    {
+        name => 'dcps',
+        file => "tests/dcps_tests.lst",
+        default => 1,
+    },
+    {
+        name => 'security',
+        file => "tests/security/security_tests.lst",
+    },
+    {
+        name => 'java',
+        file => "java/tests/dcps_java_tests.lst",
+    },
+    {
+        name => 'modeling',
+        file => "tools/modeling/tests/modeling_tests.lst",
+    },
+);
+my %builtin_test_lists_hash = map { $_->{name} => $_ } @builtin_test_lists;
+
+sub print_usage {
+    my $error = shift // 1;
+
+    my $fd = $error ? *STDERR : *STDOUT;
+    print $fd
+        "auto_run_tests.pl [<options> ...] [<list_file> ...]\n" .
```

Comment:
Why the change from using `-l` to specify list files?  This is now diverging from ACE - not necessarily a problem but it may be confusing

Reply:
ACE's already accepts positional argument list files, and we completely control this one and usage of this script, so it made sense to me to eliminate `-l` if it is no longer needed.